src/components/heading/FilterPrice.jsx

```diff
@@ -0,0 +1,154 @@
+import { TextField } from '@mui/material';
+
+function FilterPrice() {
+  const [open, setOpen] = useState(false);
+
+  const { control, setValue } = useFormContext();
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const minPrice = (price) => {
+    setValue('fromPrice', price);
+  };
+
+  const maxPrice = (price) => {
+    setValue('lastPrice', price);
+  };
+
+  const currencyItem = (curr) => {
+    setValue('currency', curr);
+  };
+
+  return (
+    <>
+      <Button
+        variant="outlined"
+        size="large"
+        onClick={handleClickOpen}
+        sx={{
+          m: 1,
+          border: '1px solid rgba(0, 0, 0, 0.25)',
+          width: '134px',
+          height: '56px',
+          color: 'rgba(0, 0, 0, 0.65)',
+        }}
+      >
+        Price
+      </Button>
+
+      <Dialog
+        disableEscapeKeyDown
+        open={open}
+        onClose={handleClose}
+      >
+        <DialogContent sx={{ pb: 0 }}>
+
+          <Box
+            component="form"
+            sx={{
+              display: 'flex', flexWrap: 'nowrap', width: '500px',
+            }}
+          >
+            <FormControl
+              sx={{ m: 1, width: '250px' }}
+            >
+              <Controller
+                name="fromPrice"
+                control={control}
+                render={({ field: { value } }) => (
+                  <TextField
+                    label="From"
+                    value={value}
+                    onChange={(event) => minPrice(event.target.value)}
+                    type="number"
+                    InputProps={{ inputProps: { min: 0, max: 10000000 } }}
+                  />
+                )}
+              />
+            </FormControl>
+
+            <FormControl sx={{ m: 1, width: '250px' }}>
+              <Controller
+                name="lastPrice"
+                control={control}
+                render={({ field: { value } }) => (
+                  <TextField
+                    label="To"
+                    variant="outlined"
+                    value={value}
+                    onChange={(event) => maxPrice(event.target.value)}
+                    type="number"
+                    InputProps={{ inputProps: { min: 0, max: 10000000 } }}
+                  />
+                )}
+              />
+            </FormControl>
+
+            <FormControl variant="outlined" sx={{ m: 1, width: '150px' }}>
+              <Controller
+                name="currency"
+                control={control}
+                render={({ field: { value } }) => (
+                  <TextField
+                    select
+                    defaultValue="BYN"
+                    variant="outlined"
+                    value={value}
+                    onChange={(event) => currencyItem(event.target.value)}
+                  >
+                    <MenuItem value={1}>BYN</MenuItem>
```

Comment:
I think you already have an const array that you can use here instead of hardcoding